cmd/nginx/main_test.go

```diff
@@ -18,46 +18,51 @@ package main
 
 import (
 	"fmt"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/ingress-nginx/internal/file"
+	"k8s.io/ingress-nginx/internal/ingress/controller"
 	"os"
 	"syscall"
 	"testing"
 	"time"
 
-	"k8s.io/ingress-nginx/internal/file"
-	"k8s.io/ingress-nginx/internal/ingress/controller"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func TestCreateApiserverClient(t *testing.T) {
-	home := os.Getenv("HOME")
-	kubeConfigFile := fmt.Sprintf("%v/.kube/config", home)
-
-	cli, err := createApiserverClient("", kubeConfigFile)
-	if err != nil {
-		t.Fatalf("Unexpected error creating Kubernetes REST client: %v", err)
-	}
-	if cli == nil {
-		t.Fatal("Expected a REST client but none returned.")
-	}
-
-	_, err = createApiserverClient("", "")
+	_, err := createApiserverClient("", "")
```

Comment:
This calls `kubernetes.NewForConfig()`, which will always fail without API. Not sure if it's really interesting to test `createApiserverClient()` at all, unless we make it testable (eg. create a `fake` client if called by tests).

Reply:
I think it will be a bigger effort to make it more testable, but maybe we can set that for a future release.

Reply:
I can look into this in the future if its ok.